src/differentiation/compute_jacobian_ad.jl

```diff
@@ -98,18 +98,37 @@ function forwarddiff_color_jacobian!(J::AbstractMatrix{<:Number},
     sparsity = jac_cache.sparsity
     color_i = 1
     chunksize = length(first(first(jac_cache.p)))
+    fill!(J, zero(eltype(J)))
 
-    for i in 1:length(p)
+    for i in eachindex(p)
         partial_i = p[i]
         t .= Dual{typeof(f)}.(x, partial_i)
         f(fx,t)
         if sparsity isa SparseMatrixCSC
             rows_index, cols_index, val = findnz(sparsity)
             for j in 1:chunksize
                 dx .= partials.(fx, j)
-                for k in 1:length(cols_index)
-                    if color[cols_index[k]] == color_i
-                        J[rows_index[k], cols_index[k]] = dx[rows_index[k]]
+                if ArrayInterface.fast_scalar_indexing(x1)
+                    for i in 1:length(cols_index)
+                        if color[cols_index[i]] == color_i
+                            if J isa SparseMatrixCSC
+                                J.nzval[i] = dx[rows_index[i]]
+                            else
+                                J[rows_index[i],cols_index[i]] = dx[rows_index[i]]
+                            end
+                        end
+                    end
+                else
+                    #=
+                    J.nzval[rows_index] .+= (color[cols_index] .== color_i) .* dx[rows_index]
+                    or
+                    J[rows_index, cols_index] .+= (color[cols_index] .== color_i) .* dx[rows_index]
+                    += means requires a zero'd out start
+                    =#
+                    if J isa SparseMatrixCSC
```

Comment:
These are tricky to read, can we use the sugared version of `get/set index`?

Reply:
Yeah, it's really tricky so I left the comment above. What's the sugared version of get/set index?

Reply:
Ah ok I just realized there is no `J.nzval.[idxs] = ...` or so, with a dotted square bracket notation.

> What's the sugared version of get/set index?

`J[i] = J[y]`

Reply:
yeah 🤷‍♂ https://github.com/JuliaLang/julia/issues/30845